addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/ZWaveConfigProvider.java

```diff
@@ -55,6 +55,8 @@
     private static Set<ThingTypeUID> zwaveThingTypeUIDList = new HashSet<ThingTypeUID>();
     private static List<ZWaveProduct> productIndex = new ArrayList<ZWaveProduct>();
 
+    private static Object productIndexLock = new Object();
```

Comment:
I would prefer to use final for such a lock object. So, you cannot reassign it...